cmd/proxy/main.go

```diff
@@ -393,7 +393,7 @@ func createFacade(
 		return nil, err
 	}
 
-	blockProc, err := process.NewBlockProcessor(connector, bp)
+	blockProc, err := process.NewBlockProcessor(connector, bp, nodeStatusProc.GetLatestFullySynchronizedHyperblockNonce)
```

Comment:
1. is there a check that the shard block is the same as the metachain notarized shard block with the same nonce?
e.g shard header hash in the metachain block is the same one for which nonce is reported by the shard.

2. Does the shard only return nonces already finalized by metachain in this method? or for example if we have latest metachain nonce 100 with shard header from shard 0 with nonce 98, then nsp.getNodeStatusMetrics(0) would return which nonce if in shard the nonce reached is also 100 but 100 is not final yet?

if we cannot confirm that the nonce returned by getNodeStatusMetrics is finalized or at least it is for the same header hash notarized in the metachain block, then I don't think we can use this.

Reply:
**1**. I'm not sure if is a check for this.
**2**. nsp.getNodeStatusMetrics(0) will return  l**atest cross notarized block header nonce** and this block should be a final metablock

@andreibancioiu maybe you have more information

Reply:
A returned hyperblock is not necessarily completely final (because the metablock isn't necessarily final, even if the shard blocks are / might be). A hyperblock is constructed by querying all the shard blocks by header hash.

@AdoAdoAdo, however, if one client of the hyperblocks API would decide to always query nonces under `erd_highest_final_nonce` of the metachain instead of querying under `erd_nonce` of the metachain (e.g. [like this](https://docs.elrond.com/querying-the-blockchain#querying-finality-information)), then everything should be all right, right?

Reply:
Furthermore, `MetricCrossCheckBlockHeight` doesn't nonce-point to a final metablock. Code from `metablock.CommitBlock()`

```
log.Info("meta block has been committed successfully" ...)
notarizedHeadersHashes, errNotCritical := mp.updateCrossShardInfo(header)
...
```